### PR TITLE
Update the AlCaRecoMatrix for Run 3

### PR DESCRIPTION
**Pull request by francescobrivio**
#### PR description:
This PR updates the AlCaRecoMatrix to be used for Run 3:
 - Moved the `AlCaRecoMatrix[2018|2017|2016|2015]` to `AlCa/doc/pastAlCaRecoMatrix.py` since they are not really used anywhere in CMSSW, but only kept for documentation purposes
 - Updated the `AlCaRecoMatrix` to the values for Run 3, as described in the [AlCaRecoMatrix twiki for 2022](https://twiki.cern.ch/twiki/bin/view/CMS/AlCaRecoMatrix#COLLISIONS_2022)

#### PR validation:
Tested with:
`runTheMatrix.py -l limited -j 16 --ibeos -t 4`

#### Backport:
Not a backport but a backport to 12_3_X will be provided soon.

### CODE REVIEW

**Comment by francescobrivio** [comment]
@cmsbuild please test

File: Configuration/AlCa/python/autoAlca.py [hunk 1]
```diff
@@ -1,87 +1,40 @@
...
+                  "AlCaLumiPixelsCountsPrompt"  : "AlCaPCCZeroBias+RawPCCProducer",
+                  # These two (AlCaPhiSym, AlCaP0) cannot run on RAW, they are just meant to run on the dedicated AlcaRAW so they do not enter the allForPrompt list
+                  "AlCaPhiSym"                  : "",
+                  "AlCaP0"                      : "",
+                  "ALCAPPS"                     : "PPSCalMaxTracks", # Express producer
+                  "AlCaPPS"                     : "PPSCalMaxTracks", # Prompt  producer
+                  "Charmonium"                  : "TkAlJpsiMuMu",
+                  "Commissioning"               : "HcalCalIsoTrk+HcalCalIsolatedBunchSelector+TkAlMinBias+SiStripCalMinBias",
+                  "Cosmics"                     : "SiPixelCalCosmics+SiStripCalCosmics+TkAlCosmics0T+MuAlGlobalCosmics",
+                  "DoubleEG"                    : "EcalCalZElectron+EcalUncalZElectron+HcalCalIterativePhiSym+HcalCalIsoTrkProducerFilter",
```
Comment (tvami) [inline Configuration/AlCa/python/autoAlca.py:12]: we dont have `EcalCalZElectron` in the twiki

Comment (tvami) [reply Configuration/AlCa/python/autoAlca.py:12]: on the other hand `EcalUncalWElectron` is  attached to`DoubleEG` in the twiki

File: Configuration/AlCa/python/autoAlca.py [hunk 2]
```diff
@@ -1,87 +1,40 @@
...
+                  "DoubleMuon"                  : "TkAlZMuMu+TkAlDiMuonAndVertex+MuAlCalIsolatedMu",
+                  "DoubleMuParked"              : "MuAlCalIsolatedMu+MuAlOverlaps+TkAlZMuMu",
```
Comment (tvami) [inline Configuration/AlCa/python/autoAlca.py:14]: Is the `DoubleMuParked` planned for Run-3?

File: Configuration/AlCa/python/autoAlca.py [hunk 3]
```diff
@@ -1,87 +1,40 @@
...
+                  "EGamma"                      : "EcalESAlign+EcalUncalWElectron+EcalUncalZElectron+HcalCalIsoTrkProducerFilter+HcalCalIterativePhiSym",
+                  "Express"                     : "SiStripCalZeroBias+TkAlMinBias+SiStripPCLHistos+SiStripCalMinBias+SiStripCalMinBiasAAG+Hotline+SiPixelCalZeroBias",
```
Comment (tvami) [inline Configuration/AlCa/python/autoAlca.py:16]: Is this `Hotline` also planned?

File: Configuration/AlCa/python/autoAlca.py [hunk 4]
```diff
@@ -1,87 +1,40 @@
...
+                  "HcalNZS"                     : "HcalCalMinBias",
+                  "HLTPhysics"                  : "TkAlMinBias",
+                  "JetHT"                       : "HcalCalIsoTrkProducerFilter",
+                  "MET"                         : "HcalCalNoise",
+                  "MinimumBias"                 : "SiStripCalZeroBias+SiStripCalMinBias+TkAlMinBias",
+                  "MuOnia"                      : "TkAlUpsilonMuMu",
+                  "MuOniaParked"                : "TkAlJpsiMuMu+TkAlUpsilonMuMu",
+                  "NoBPTX"                      : "TkAlCosmicsInCollisions",
+                  "SingleElectron"              : "EcalUncalWElectron+EcalUncalZElectron+HcalCalIterativePhiSym+EcalESAlign",
+                  "SingleMuon"                  : "SiPixelCalSingleMuonLoose+SiPixelCalSingleMuonTight+TkAlMuonIsolated+MuAlCalIsolatedMu+MuAlOverlaps+MuAlZMuMu+HcalCalHO+HcalCalIterativePhiSym+HcalCalHBHEMuonProducerFilter",
```
Comment (tvami) [inline Configuration/AlCa/python/autoAlca.py:28]: I think `MuAlOverlaps` is to be removed, the twiki says "
Removed MuAlOverlaps and MuAlZMuMu from SingleMuon and DoubleMuon datasets (in agreement with Muon Alignment group)"

File: Configuration/AlCa/python/autoAlca.py [hunk 5]
```diff
@@ -1,87 +1,40 @@
...
+                  "SinglePhoton"                : "HcalCalGammaJet",
```
Comment (tvami) [inline Configuration/AlCa/python/autoAlca.py:29]: Isnt the `SinglePhoton` now part of the `EGamma`?

Comment (francescobrivio) [reply Configuration/AlCa/python/autoAlca.py:29]: Probably yes, but where was this announced?

File: Configuration/AlCa/python/autoAlca.py (hunk 3 above)
Comment (francescobrivio) [reply Configuration/AlCa/python/autoAlca.py:16]: I didn't find what this `Hotline` is, so I left it there for the moment...

Comment (tvami) [reply Configuration/AlCa/python/autoAlca.py:16]: Hi @cms-sw/xpog-l2 I understand that his `Hotline` is a bunch of skims. Is this still relevant for Run-3? Thanks!

File: Configuration/AlCa/python/autoAlca.py (hunk 5 above)
Comment (mmusich) [reply Configuration/AlCa/python/autoAlca.py:29]: Was already so in 2018 IIRC

Comment (francescobrivio) [reply Configuration/AlCa/python/autoAlca.py:29]: Indeed `SinglePhoton` and `SingleElectron` were merged in `EGamma` in 2018! Thanks to both!

File: Configuration/AlCa/python/autoAlca.py (hunk 3 above)
Comment (kskovpen) [reply Configuration/AlCa/python/autoAlca.py:16]: @sam7k9621, do you have some recollection of these skims?

Comment (mmusich) [reply Configuration/AlCa/python/autoAlca.py:16]: For the record (info has already been shared with @tvami privately):
   * Group responsible of the `HotLine`: https://twiki.cern.ch/twiki/bin/viewauth/CMS/DatasetDefinitionTeam
   * details on the wf: https://indico.cern.ch/event/377707/

My impression is that this has been fully replace by Scouting.

File: Configuration/AlCa/python/autoAlca.py (hunk 1 above)
Comment (francescobrivio) [reply Configuration/AlCa/python/autoAlca.py:12]: fixed

File: Configuration/AlCa/python/autoAlca.py (hunk 3 above)
Comment (mmusich) [reply Configuration/AlCa/python/autoAlca.py:16]: > I'll remove the Hotline then

Just to be on the safe side, I think you'd better check with EXO / PPD

File: Configuration/AlCa/python/autoAlca.py (hunk 1 above)
Comment (francescobrivio) [reply Configuration/AlCa/python/autoAlca.py:12]: just FYI `EcalCalZElectron` is not present in the tier0 matrix, but until we get a confirmation from ECAL wethere this is to be kept or not, I don't see any problem in leaving it here for the moment.

File: Configuration/AlCa/python/autoAlca.py (hunk 2 above)
Comment (rappoccio) [reply Configuration/AlCa/python/autoAlca.py:14]: Will have to discuss after XC meeting 21-April.

Comment (francescobrivio) [reply Configuration/AlCa/python/autoAlca.py:14]: Given that there was not clear outcome at the meeting on the parking datasets that will be produced, I prefer to leave this unouched for the moment.

File: Configuration/AlCa/python/autoAlca.py (hunk 3 above)
Comment (francescobrivio) [reply Configuration/AlCa/python/autoAlca.py:16]: IIUC from a private mail thread with @lowette and @zdemirag , EXO needs more time to assess if the `Hotline` is needed or not. So for the moment I will leave this untouched.

Comment (mmusich) [reply Configuration/AlCa/python/autoAlca.py:16]: Just want to point out that it was not produced in 2018:
```
 dasgoclient -query='dataset dataset=/*Express/Run*Hotline*/* status=*'
/StreamExpress/Run2015B-Hotline-Express-v1/ALCARECO
/StreamExpress/Run2015C-Hotline-Express-v1/ALCARECO
/StreamExpress/Run2015D-Hotline-Express-v3/ALCARECO
/StreamExpress/Run2015D-Hotline-Express-v4/ALCARECO
/StreamExpress/Run2015E-Hotline-Express-v1/ALCARECO
/StreamExpress/Run2016A-Hotline-Express-v1/ALCARECO
/StreamExpress/Run2016A-Hotline-Express-v2/ALCARECO
/StreamExpress/Run2016B-Hotline-Express-v1/ALCARECO
/StreamExpress/Run2016B-Hotline-Express-v2/ALCARECO
/StreamExpress/Run2016C-Hotline-Express-v2/ALCARECO
/StreamExpress/Run2016D-Hotline-Express-v2/ALCARECO
/StreamExpress/Run2016E-Hotline-Express-v2/ALCARECO
/StreamExpress/Run2016F-Hotline-Express-v1/ALCARECO
/StreamExpress/Run2016G-Hotline-Express-v1/ALCARECO
/StreamExpress/Run2016H-Hotline-Express-v1/ALCARECO
/StreamExpress/Run2016H-Hotline-Express-v2/ALCARECO
/StreamExpress/Run2017A-Hotline-Express-v1/ALCARECO
/StreamExpress/Run2017A-Hotline-Express-v2/ALCARECO
/StreamExpress/Run2017A-Hotline-Express-v3/ALCARECO
/StreamExpress/Run2017B-Hotline-Express-v1/ALCARECO
/StreamExpress/Run2017B-Hotline-Express-v2/ALCARECO
/StreamExpress/Run2017C-Hotline-Express-v1/ALCARECO
/StreamExpress/Run2017C-Hotline-Express-v2/ALCARECO
/StreamExpress/Run2017C-Hotline-Express-v3/ALCARECO
/StreamExpress/Run2017D-Hotline-Express-v1/ALCARECO
/StreamExpress/Run2017E-Hotline-Express-v1/ALCARECO
/StreamExpress/Run2017F-Hotline-Express-v1/ALCARECO
```

**Comment by francescobrivio** [comment]
@cmsbuild please test

**Comment by francescobrivio** [comment]
@smuzaffar seems like tests are stuck for this PR?

**Comment by smuzaffar** [comment]
please test
let restart

**Comment by tvami** [comment]
+alca
 * changes are according to the twiki (and the subsystem needs)
 * Jenkins tests pass

**Comment by tvami** [comment]
hold
 * there is a question regarding `TkAlMinBias` in `JetHT`

**Comment by tvami** [comment]
unhold

**Comment by tvami** [comment]
@cmsbuild , please test

**Comment by tvami** [comment]
+alca

**Comment by perrotta** [comment]
+1
- Adjustments seem settled down now...

**Comment by makortel** [comment]
This PR might have caused 1001.2 step2 to fail with
```
----- Begin Fatal Exception 26-Apr-2022 13:15:01 CEST-----------------------
An exception of category 'ProductNotFound' occurred while
   [0] Processing  Event run: 305064 lumi: 2 event: 1783800 stream: 0
   [1] Running path 'pathALCARECOAlCaPCCRandom'
   [2] Calling method for module AlcaPCCIntegrator/'alcaPCCIntegratorRandom'
Exception Message:
Principal::getByToken: Found zero products matching all criteria
Looking for type: reco::PixelClusterCountsInEvent
Looking for module label: hltAlcaPixelClusterCounts
Looking for productInstanceName: alcaPCCEvent

   Additional Info:
      [a] If you wish to continue processing events after a ProductNotFound exception,
add "SkipEvent = cms.untracked.vstring('ProductNotFound')" to the "options" PSet in the configuration.

----- End Fatal Exception -------------------------------------------------
```
see #37685.

**Pull request merged**